### PR TITLE
Add visible labels to resource select component and filter dialog

### LINKED ISSUE #3396: No visible label provided for 'Field', 'Condition' and 'Value' combo box on 'Add Filter' pane.: A11y_.NetAspireDashboard_StructuredLog_Add Filter_LabelsorInstructions
"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**

### Tags:
#A11yTCS; #A11yMAS; #A11ySev2; #FTP; #Win11; #ChromiumEdge; #FTP; #DesktopWeb; #.NET Core; #SH_.Net Aspire Dashboard_Web_Mar2024; #WCAG3.3.2; #LabelsorInstruction;

### Environment Details:
Application Name: .Net Aspire Dashboard
URL: https://localhost:19888/
Chrome Version 122.0.2365.80 (Official build) (64-bit)
Window Version: Win 11
OS Build: 22621.3085

### Repro Steps:

1. Hit the URL: https://localhost:15887/ and login with valid credentials.
2. TAB to "Structured Log" tab item and hit ENTER key.
3. TAB to "Filter" and hit ENTER key.
4. Observe that no visible label provided for 'Field', 'Condition' and 'Value' combo box on 'Add Filter' pane.

### Actual Results:
No visible label provided for 'Field', 'Condition' and 'Value' combo box on 'Add Filter' pane.

**Same issue repro on 'Structured log' page -> 'Select application' combo box.**

### Expected Results:
Visible label should be provided for 'Field', 'Condition' and 'Value' combo box on 'Add Filter' pane.

### User Impact:
Screen reader users will not be able to understand the purpose of the dropdown and will miss the information about it.

### Attachment:
3396_A11y_ NetAspireDashboard_StructuredLog_Add Filter_LabelsorInstructions

StructuredLogs_SelectApplication_Label

### PR DESCRIPTION
**Pull request by adamint**
Fixes #3396, we need to have visible labels for interactive components. We also had several kinds of All/None options, so I standardized these into (All) and (None)

I added labels to the inputs in FilterDialog and made them the same width.
image

The resource select also has a label to its left.
image

I removed the label to the right of the resource select on the console log page when there is no resource selected, as it just mentioned that there was no resource selected, which is redundant with the select's (None) selected text.

image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3727)

### CODE REVIEW

**Comment by adamint** [comment]
@tlmii @JamesNK

**Comment by JamesNK** [comment]
"Select a resource" is slightly higher than center. For some reason CSS adds a small margin:
image

Override the CSS when inside the toolbar to have no margin top or bottom.

image

Structured logs page has "Level:" with a colon. Should stay consistent with that. Add colon after "Select a resource".

Also, the "Level:" here isn't a label. It should probably be.

**Comment by adamint** [comment]
@JamesNK I just want to confirm that !important is necessary since fluentui-blazor uses a class and attribute selector (.fluent-input-label[attribute])? If not, do you know how to avoid !important here?

**Comment by tlmii** [comment]
> @JamesNK I just want to confirm that !important is necessary since fluentui-blazor uses a class and attribute selector (.fluent-input-label[attribute])? If not, do you know how to avoid !important here?

Making the selector `.layout fluent-toolbar > .fluent-input-label` would give it a specificity of 21, beating the built-in rule's specificity of 20. Seems like a superfluous addition, but it does make a difference and avoids the `!important` which always feels like a code smell.

That misalignment feels like its a bug in the library, though, since we're just using the Label attribute of the FluentSelect component and its generating the rest. @vnbaaij  thoughts on that?

**Comment by vnbaaij** [comment]
> Structured logs page has "Level:" with a colon. Should stay consistent with that. Add colon after "Select a resource".
> 
> Also, the "Level:" here isn't a label. It should probably be.

In the Fluent 2 guides (https://fluent2.microsoft.design/components/web/react/field/usage#layout) no colons are used for labels. I would rather remove them instead of adding them.

**Comment by adamint** [comment]
Removed colons @vnbaaij, @tlmii switched selectors

**Comment by vnbaaij** [comment]
> That misalignment feels like its a bug in the library, though, since we're just using the Label attribute of the FluentSelect component and its generating the rest. @vnbaaij thoughts on that?

Going with the Fluent 2 desing, we assume the label is placed above the component it belongs to. We need that margin-bottom to prevent things from getting too cramped: 
image

What I started doing based on this issue, was adding a `Orientation` parameter to the `FluentInputLabel` (which is the component being rendered when you set the `Label` on the `FluentSelect`) that will add a 'fluent-input-label-horizontal' class (setting `margin-bottom: 0;`) in the case of setting it to `Orientation.Horizontal`.
Now, while that is a nice addition, it has no use in this case as you are setting the Label property on the Select and there is no way to specify the orientation.
When this addition is included in the library, you could remove the `Label` parameter from the Select and add a `FluentInputLabel` with a `Orientation="Orientation.Horizontal"` manually (inside a `FluentStack`)

**Comment by vnbaaij** [comment]
PR: https://github.com/microsoft/fluentui-blazor/pull/1994

**Comment by adamint** [comment]
@vnbaaij @JamesNK resurrecting this after the last pr got merged. Same changes (after prior feedback applied), and uses the same (All) and (None) strings everywhere.
 
image
image
image
image
image

**Comment by vnbaaij** [comment]
@adamint sorry, but I don't understand what exactly is the issue now (going on these screenshots). Can you clarify?

**Comment by adamint** [comment]
@vnbaaij #3396 is still active, this PR adds labels to the filter dialog input components, as well as the resource select

**Comment by adamint** [comment]
@JamesNK as well

**Comment by JamesNK** [comment]
Change "Select a resource" to "Resource" on all the pages:
image

This is consistent with other UI in the page.

Remove "Value" placeholder:
image

**Comment by adamint** [comment]
> Change "Select a resource" to "Resource" on all the pages: image This is consistent with other UI in the toolbar, e.g. there is just the text "Filter", not "Select a filter".
> 
> Remove "Value" placeholder: image The other fields here are pre-selected. It could be confusing that value is also pre-selected.
> 
> I think after these changes this is ready to merge.

Changed both.

File: src/Aspire.Dashboard/Resources/ControlsStrings.resx [hunk 1]
```diff
@@ -255,7 +255,7 @@
     <value>No metrics data found</value>
   </data>
   <data name="SelectAResource" xml:space="preserve">
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Resources/ControlsStrings.resx:257]: SelectAResource doesn't make sense anymore. How about ResourceLabel.

**Review by JamesNK — APPROVED** [review]

**Pull request merged**